### PR TITLE
tests: remove core18 if installed by tests

### PR DESCRIPTION
**Pull request by zyga**
When tests run on ubuntu core, the automatic cleanup performed is
different and imperfect. In particular it is very easy to "leak"
additional base snaps, like core18.

Before we have a better mechanism available, remove core18 while
restoring a test case that did not have core18 before the test executed.

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>

### CODE REVIEW

File: tests/lib/prepare-restore.sh [hunk 1]
```diff
@@ -543,6 +553,13 @@ restore_suite_each() {
         rm -rf "${PWD}.tar"
     fi
 
+    # The core18 snap may have been automatically installed by the test.
+    # If it was not present before the test then it should be removed.
+    if [ -e /run/spread/please-remove-core18 ]; then
```
Comment (anonymouse64) [inline tests/lib/prepare-restore.sh]: should this maybe be:

```suggestion
    if [ -f /run/spread/please-remove-core18 ]; then
```

I don't know why this file would "exist" but not be a regular file, but -f seems more natural to me.

**Comment by zyga** [comment]
This is red because on some systems we still pull in more test snaps. I'm going though this list now.

**Comment by zyga** [comment]
I think there are two aspects that make this fail:

- There are still some tests that keep a `base: core18` snap installed, which is not automatically removed on core systems.
- The leak detector is blocked by several other bugs which means it cannot be merged to easily identify the offending tests

In addition some tests leave snapd removed/disabled which makes the remove part fail. I need to re-consider how to approach this.

**Comment by zyga** [comment]
The core18 situation should now be resolved, this might now pass.

**Comment by zyga** [comment]
I don't think this is really necessary now. With the fix to snap removal this is redundant.

**Pull request closed without merging**